File: frontend/src/api/task.ts
import { request } from './request'

/**
 * 任务查询参数
 */
export interface TaskQuery {
  assignee?: string
  candidateUser?: string
  candidateGroup?: string
  processKey?: string
  processInstanceId?: string
  businessKey?: string
  taskName?: string
  taskStatus?: 'todo' | 'done'
  pageNum?: number
  pageSize?: number
}

/**
 * 任务
 */
export interface Task {
  taskId: string
  taskName: string
  taskKey: string
  processInstanceId: string
  processDefinitionId?: string
  processKey?: string
  businessKey?: string
  assignee?: string
  owner?: string
  createTime: string
  endTime?: string
  status: string
  priority?: number
  description?: string
  formKey?: string
  variables?: Record<string, any>
}

/**
 * 完成任务参数
 */
export interface CompleteTaskParams {
  taskId: string
  assignee: string
  comment?: string
  variables?: Record<string, any>
}

/**
 * 分页结果
 */
export interface PageResult<T> {
  list: T[]
  total: number
  pageNum: number
  pageSize: number
  totalPages: number
}

/**
 * 任务管理相关 API
 */
export const taskApi = {
  /**
   * 查询任务列表
   */
  queryTasks(params: TaskQuery): Promise<{ data: PageResult<Task> }> {
    return request.get('/tasks/list', { params })
  },

  /**
   * 查询单个任务
   */
  getTask(taskId: string): Promise<{ data: Task }> {
    return request.get(`/tasks/${taskId}`)
  },

  /**
   * 完成任务
   */
  completeTask(params: CompleteTaskParams): Promise<any> {
    return request.post('/tasks/complete', params)
  },

  /**
   * 认领任务
   */
  claimTask(taskId: string, userId: string): Promise<any> {
    return request.post(`/tasks/${taskId}/claim`, null, {
      params: { userId }
    })
  },

  /**
   * 委派任务
   */
  delegateTask(taskId: string, delegateUserId: string): Promise<any> {
    return request.post(`/tasks/${taskId}/delegate`, null, {
      params: { delegateUserId }
    })
  },

  /**
   * 转办任务
   */
  transferTask(taskId: string, targetUserId: string): Promise<any> {
    return request.post(`/tasks/${taskId}/transfer`, null, {
      params: { targetUserId }
    })
  }
}

File: backend/src/main/java/com/bank/workflow/domain/task/entity/Task.java
package com.bank.workflow.domain.task.entity;

import lombok.Data;

import java.util.Date;
import java.util.Map;

/**
 * 任务领域实体
 *
 * @author Workflow Team
 * @since 1.0.0
 */
@Data
public class Task {
    
    /** 任务ID */
    private String taskId;
    
    /** 任务名称 */
    private String taskName;
    
    /** 任务定义KEY */
    private String taskKey;
    
    /** 流程实例ID */
    private String processInstanceId;
    
    /** 流程定义ID */
    private String processDefinitionId;
    
    /** 流程定义KEY */
    private String processKey;
    
    /** 业务KEY */
    private String businessKey;
    
    /** 任务办理人 */
    private String assignee;
    
    /** 任务所有者 */
    private String owner;
    
    /** 候选用户 */
    private String candidateUsers;
    
    /** 候选组 */
    private String candidateGroups;
    
    /** 任务创建时间 */
    private Date createTime;
    
    /** 任务完成时间 */
    private Date endTime;
    
    /** 任务状态 */
    private String status;
    
    /** 任务优先级 */
    private Integer priority;
    
    /** 任务描述 */
    private String description;
    
    /** 表单KEY */
    private String formKey;
    
    /** 任务变量 */
    private Map<String, Object> variables;
}

